### PR TITLE
Create CBD-Article component

### PR DESCRIPTION
**Pull request by tahmeedshah**
### General description
_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs
_Link to the related design prototypes (if applicable)_

### Testing instructions
_Provide minimal instructions on how to test this PR._

* Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
* [ ]  Branch name / PR includes the related Jira ticket Id.
* [ ]  Tests to check core implementation / bug fix added.
* [ ]  All checks in Continuous Integration workflow pass.
* [ ]  Feature functionally tested by reviewer(s).
* [ ]  Code reviewed by reviewer(s).
* [ ]  Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: app/views/portals/cbd-article.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    const auth = useAuth();
+    const { t } = useI18n({ messages });
+    const articlesApi = new ArticlesApi({tokenReader:()=>auth.token()});
+
+    const emit = defineEmits(['load']);
+
+    const props = defineProps({
+        query: {
+            type: Object,
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:35]: Object or Array

File: app/views/portals/cbd-article.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    import messages from '../../app-text/components/kb.json';
+    import { lstring } from '../../components/kb/filters';
+    import { useAuth } from '@scbd/angular-vue/src/index.js';
+    import ArticlesApi from '../../components/kb/article-api';
+
+    const auth = useAuth();
+    const { t } = useI18n({ messages });
+    const articlesApi = new ArticlesApi({tokenReader:()=>auth.token()});
+
+    const emit = defineEmits(['load']);
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:31]: onArticleLoad

File: app/views/portals/cbd-article.vue [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    const props = defineProps({
+        query: {
+            type: Object,
+            required: false
+        }
+    });
+
+    const loading = ref(true);
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:40]: default should be false

File: app/views/portals/cbd-article.vue [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    const article = ref({})
+
+    onMounted( async ()=>{
+        const articleResult = await articlesApi.queryArticles(props.query);
+        if(articleResult.length > 0){
+            article.value = articleResult[0] ;
+            emit('load', article.value);
+            loading.value = false;
+        }
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:49]: try catch will be good here

File: app/views/portals/cbd-article.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<template>
+    <div class="loading" v-if="loading">
+        <i class="fa fa-cog fa-spin fa-lg"></i> 
+        {{ t("loading") }}...
+    </div>
+    <div v-if="!loading">
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:6]:
```suggestion
    <div v-if="!loading && article">
```

File: app/views/portals/cbd-article.vue [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        <link type="text/css" rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@scbd/ckeditor5-build-inline-full@35.0.0/build/content-style.css">
+
+        <div v-if="article?.coverImage">
+            <img v-bind:src="getSizedImage(article?.coverImage.url, '800x800')">
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:10]: where is the definition for getSizedImage?

File: app/views/portals/article-id.vue [hunk 7]
```diff
@@ -10,7 +10,7 @@
 <script>
 
 import { mapObjectId, isObjectId } from '~/api/api-base.js';
-import { cbdArticle } from 'scbd-common-articles';
+import  cbdArticle  from './cbd-article.vue';
 
 export default {
```
Comment (blaisefonseca) [inline app/views/portals/article-id.vue:15]: why is this option

File: app/api/api-base.js [hunk 8]
```diff
@@ -140,7 +140,7 @@ export function stringifyUrlParams(params){
 
 //ToDo: can we remove this ?
 export async function defaultTokenReader() {
-  const auth = useAuth();
-  const userToken = await auth.token();
-  return userToken;
+  // const auth = useAuth();
+  // const userToken = await auth.token();
+  // return userToken;
 }
```
Comment (blaisefonseca) [inline app/api/api-base.js:146]: delete if not required

**Review by blaisefonseca — COMMENTED** [review]
check comments

File: app/views/portals/cbd-article.vue (hunk 1 above)
Comment (tahmeedshah) [reply app/views/portals/cbd-article.vue:35]: Object,
 {"ag":"[{\"$match\":{\"_id\":{\"$oid\":\"64d6700c8e4a59a51e79dff2\"}}}]"}

File: app/api/api-base.js (hunk 8 above)
Comment (tahmeedshah) [reply app/api/api-base.js:146]: article-id.vue is in vue2. in next PR it is converted to Vue3.

File: app/views/portals/article-id.vue (hunk 7 above)
Comment (tahmeedshah) [reply app/views/portals/article-id.vue:15]: article-id.vue is in vue2. in next PR it is converted to Vue3.

File: app/views/portals/cbd-article.vue (hunk 6 above)
Comment (tahmeedshah) [reply app/views/portals/cbd-article.vue:10]: added, but need to move to some common file.

File: app/views/portals/cbd-article.vue (hunk 5 above)
Comment (tahmeedshah) [reply app/views/portals/cbd-article.vue]: done

File: app/views/portals/cbd-article.vue [hunk 9]
```diff
@@ -28,25 +28,39 @@
...
+        try{
+            loading.value = true;
+            const articleResult = await articlesApi.queryArticles(props.query);
+            if(articleResult.length > 0){
+                article.value = articleResult[0] ;
+                emit('onArticleLoad', article.value);
+            }
+        } 
+        catch (err) {
+            console.log(err);
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:53]: can we give feedback to user, there was an error?

File: app/views/portals/cbd-article.vue [hunk 10]
```diff
@@ -0,0 +1,75 @@
+<template>
+    <div class="loading" v-if="loading">
+        <i class="fa fa-cog fa-spin fa-lg"></i> 
+        {{ t("loading") }}...
+    </div>
+    <div v-if="!loading && article">
+        <link type="text/css" rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@scbd/ckeditor5-build-inline-full@35.0.0/build/content-style.css">
+
+        <div v-if="article?.coverImage">
+            <img v-bind:src="getSizedImage(article?.coverImage.url, '800x800')">
```
Comment (blaisefonseca) [inline app/views/portals/cbd-article.vue:10]: should be computed property

File: app/views/portals/forum-id.vue [hunk 11]
```diff
@@ -4,7 +4,8 @@
     <h1 v-if="!article && forum">{{ lstring(forum.title) }}</h1>
 
     <cbd-article :query="articleQuery" v-if="articleQuery" :hide-cover-image="true" :show-edit="true"
-      @load="onArticleLoad($event)" :admin-tags="articleAdminTags">
+      @onArticleLoad="loadArticle($event)" :admin-tags="articleAdminTags">
```
Comment (blaisefonseca) [inline app/views/portals/forum-id.vue:7]:
```suggestion
      @on-article-load="loadArticle($event)" :admin-tags="articleAdminTags">
```

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**